Content.Server/GameObjects/Components/Kitchen/KitchenMicrowaveComponent.cs

```diff
@@ -403,5 +407,27 @@ private void ClickSound()
 
         }
 
+        public SuicideKind Suicide(IEntity victim, IChatManager chat)
+        {
+            victim.TryGetComponent<BodyManagerComponent>(out var bodyManagerComponent);
+            if (bodyManagerComponent != null)
+            {
+                var heads = bodyManagerComponent.GetBodyPartsOfType(BodyPartType.Head);
+                foreach (var head in heads)
+                {
+                    var droppedHead = bodyManagerComponent.DisconnectBodyPart(head, true);
+                    _storage.Insert(droppedHead);   
+                }
+                chat.EntityMe(victim, Loc.GetString("is trying to cook {0:their} " + (heads.Count() > 1 ? "heads!": "head!"), victim));
+            } else
+            {
+                chat.EntityMe(victim, Loc.GetString("is trying to cook {0:their} head!", victim));
+            }
```

Comment:
You can refactor by making something like this, to remove the duplication of `chat.EntityMe(/* ... */)`
```c#
int headCount = 1;
if (victim.TryGetComponent<BodyManagerComponent>(out var bodyManagerComponent))
{
    // chop-off head, change headCount
}

chat.EntityMe(/* ... */);
```